### ISSUE TITLE
"Hello, NAS!" tutorial perfomance issue with zero cost model evaluation function

### ISSUE THREAD

**Issue by miracle-111** [microsoft/nni#5261]
I run "Hello, NAS!" tutorial from documentation in GPU jupyter notebook (kaggle and colab with same results). But I changed evaluate_model function as:
```
def evaluate_model(model_cls): 
    accuracy = 0.57
    nni.report_final_result(accuracy)
```
Using this evaluation function I expected instant experement execution, but it was running for 1 minute 20 seconds. When I changed max_trial_number from 4 to 8, the experement was running 2 minutes and 30 seconds...

Full code:

```
!pip install pytorch-lightning
!pip install nni
```
```
import torch.nn.functional as F
import nni.retiarii.nn.pytorch as nn
import nni.retiarii.strategy as strategy
from nni.retiarii import model_wrapper
from nni.retiarii.evaluator import FunctionalEvaluator
from nni.retiarii.experiment.pytorch import RetiariiExperiment, RetiariiExeConfig
import nni

class DepthwiseSeparableConv(nn.Module):
    def __init__(self, in_ch, out_ch):
        super().__init__()
        self.depthwise = nn.Conv2d(in_ch, in_ch, kernel_size=3, groups=in_ch)
        self.pointwise = nn.Conv2d(in_ch, out_ch, kernel_size=1)

    def forward(self, x):
        return self.pointwise(self.depthwise(x))

@model_wrapper
class ModelSpace(nn.Module):    
    def __init__(self):        
        super().__init__()
        self.conv1 = nn.Conv2d(1, 32, 3, 1)        
        self.conv2 = nn.LayerChoice([
            nn.Conv2d(32, 64, 3, 1),
            DepthwiseSeparableConv(32, 64)
        ])
        
        self.dropout1 = nn.Dropout(nn.ValueChoice([0.25, 0.5, 0.75]))
        self.dropout2 = nn.Dropout(0.5)
        feature = nn.ValueChoice([64, 128, 256])
        self.fc1 = nn.Linear(9216, feature)
        self.fc2 = nn.Linear(feature, 10)

    def forward(self, x):        
        x = F.relu(self.conv1(x))
        x = F.max_pool2d(self.conv2(x), 2)
        x = torch.flatten(self.dropout1(x), 1)
        x = self.fc2(self.dropout2(F.relu(self.fc1(x))))
        output = F.log_softmax(x, dim=1)
        return output

model_space = ModelSpace()
search_strategy = strategy.Random(dedup=True)

def evaluate_model(model_cls):    
    accuracy = 0.57
    nni.report_final_result(accuracy)
    
evaluator = FunctionalEvaluator(evaluate_model)

exp = RetiariiExperiment(model_space, evaluator, [], search_strategy)
exp_config = RetiariiExeConfig('local')
exp_config.experiment_name = 'mnist_search'
exp_config.max_trial_number = 4 
exp_config.trial_concurrency = 1
exp_config.trial_gpu_number = 1
exp_config.training_service.use_active_gpu = True
```

```
%%time
exp.run(exp_config, 8081)
```

**Comment by ultmaster** [comment]
A large part of NNI relies on a 5-second heartbeat, including GPU scheduling, metrics collection, message passing and etc. We expect about 15 seconds of overhead for each trial on GPU. Plus some additional overheads, it approximately adds up to your 1min 20secs.

We are actively addressing this issue by rewriting the core of NNI right now. Hopefully we can greatly improve the performance upon the release of NNI 3.0.

**Comment by miracle-111** [comment]
> A large part of NNI relies on a 5-second heartbeat, including GPU scheduling, metrics collection, message passing and etc. We expect about 15 seconds of overhead for each trial on GPU. Plus some additional overheads, it approximately adds up to your 1min 20secs.
> 
> We are actively addressing this issue by rewriting the core of NNI right now. Hopefully we can greatly improve the performance upon the release of NNI 3.0.

Can we expect the overhead of running an experiment to be less than 10 ms in the future versions? To find the network architecture, I use algorithms for accuracy estimation without training. Typical time to estimate accuracy of already created neural network object with random initialized weights is less than 100 ms, and number of calls when we use random or evollutional search is more than 1000. Using nni framework with current perfomance is compleatly no sense.
